joomla/joomla-cms
# [4.0-dev] Fixing SCSS issues with division deprecation and math.div()

Pull Request for PR #35639 .

# Important
This PR could be merged into [4.1-dev] as well.

### Summary of Changes

* The fontawesome icon library haven't marked their `$fa-fw-width` variable with `!default`, see [#17482](https://github.com/FortAwesome/Font-Awesome/issues/17482) which uses `(20em / 16)` expression which is deprecated in dart-sass 2.0 and also [largely reported](https://github.com/FortAwesome/Font-Awesome/issues/17946) in the fontawesome issues.

* Other changes in this PR also address other dart-sass issues, but we don't replace for instance `24 / 16` with `math.div(24, 16)` as recommended by the compiler, but with `24 * 0.0625` (which is the result of 1/16) and the reason for that is the libsass compiler still largely used and this approach is also widely adopted. I suspect scssPhp also might not have `math` implemented.

* See [this PR in fontawesome](https://github.com/FortAwesome/Font-Awesome/pull/17940) for more info. Briefly, Bootstrap core devs also continue to support libsass, so should we.

### Testing Instructions
Check SASS compiler output. You should see all SCSS compiled. If [the npm build](https://ci.joomla.org/joomla/joomla-cms/47619/1/7) is green, there is no need to test anything else.

### Actual result BEFORE applying this Pull Request
The SCSS compiler just won't do division via the `/` (forward slash) operator, the build will simply fail
https://ci.joomla.org/joomla/joomla-cms/47599


### Expected result AFTER applying this Pull Request
All SCSS files compiled successful.


### Documentation Changes Required
No.
